src/ApiService/ApiService/onefuzzlib/VmExtensionWrapper.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Azure.Core;
+using Azure.ResourceManager.Compute;
+
+namespace Microsoft.OneFuzz.Service {
+    public class VMExtensionWrapper {
+        public AzureLocation? Location { get; init; }
+        public string? Name { get; init; }
+        public string? TypePropertiesType { get; init; }
+        public string? Publisher { get; init; }
+        public string? TypeHandlerVersion { get; init; }
+        public string? ForceUpdateTag { get; init; }
+        public bool? AutoUpgradeMinorVersion { get; init; }
+        public bool? EnableAutomaticUpgrade { get; init; }
+        public BinaryData? Settings { get; init; }
+        public BinaryData? ProtectedSettings { get; init; }
+
+        public (string, VirtualMachineExtensionData) GetAsVirtualMachineExtension() {
```

Comment:
You can use an implicit [conversion instead](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/user-defined-conversion-operators).

just suggestion, feel free to ignore